test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py

```diff
@@ -0,0 +1,90 @@
+
+import functools
+import itertools
+import sys
+import torch
+
+from torch import distributed as dist
+from torch.distributed.fsdp.fully_sharded_data_parallel import ShardingStrategy
+from torch.distributed.fsdp import MixedPrecision, CPUOffload, BackwardPrefetch
+from torch.testing._internal.common_fsdp import subtest_name, FSDPInitMode, NestedWrappedModule, FSDPTest
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_utils import instantiate_parametrized_tests, parametrize, TEST_WITH_DEV_DBG_ASAN
+
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+params = "cpu_offload,backward_prefetch,sharding_strategy,mixed_precision"
+cpu_offload_config = [CPUOffload(offload_params=True), CPUOffload(offload_params=False)]
+backward_prefetch_config = [BackwardPrefetch.BACKWARD_PRE, BackwardPrefetch.BACKWARD_POST, None]
```

Comment:
In the interest of TTS, I'm pretty sure we can omit backward prefetch tests as I don't expect them to have implication for sharded grad scaling. cc @zhaojuanmao just in case I'm missing something